### PR TITLE
bug fix and small refactor

### PR DESCRIPTION
**Pull request by shawr**
fixed and symbol bug, and re factored reports and replies examples to be more flexible

### CODE REVIEW

File: examples_replies.php [hunk 1]
```diff
@@ -46,36 +46,48 @@
 // Create new MMSoap class
 $soap = new MMSoap($username, $password, $options);
 
-echo "** Send message to $recipient\n";
-// send the message
+echo "** Do you want to send a message to $recipient, y followed by ENTER to send, or ENTER to skip.\n";
+$handle = fopen("php://stdin", "r");
+$line = fgets($handle);
+if ($line == "y\n") {
```
Comment (afmm) [inline examples_replies.php]: Do we care about case sensitivity?

File: examples_replies.php [hunk 2]
```diff
@@ -46,36 +46,48 @@
...
+    $result = $response->getResult ();
+    echo $result->sent . ' sent / ' . $result->scheduled . ' scheduled / ' . $result->failed . " failed\n\n";
 
-// message has been sent, give the user time to reply
-echo "Please reply to the message you just received and hit ENTER.\n";
-$handle = fopen ( "php://stdin", "r" );
-$line = fgets ( $handle );
+    // message has been sent, give the user time to reply
+    echo "Please reply to the message you just received and hit ENTER.\n";
+    $handle = fopen("php://stdin", "r");
```
Comment (afmm) [inline examples_replies.php]: Could these two lines be extracted to a function of their own? This is the second use of this exact pattern?

Maybe, $input = getUserInput()?

File: examples_replies.php [hunk 3]
```diff
@@ -46,36 +46,48 @@
...
+if ($line == "y\n") {
+    // send the message
 
-$response = $soap->sendMessage ( $recipient, "messagemedia-php: please reply" );
-if ($response instanceof SoapFault) {
-	exit ( 'Error: ' . $response->getMessage () );
-}
-$result = $response->getResult ();
-echo $result->sent . ' sent / ' . $result->scheduled . ' scheduled / ' . $result->failed . " failed\n\n";
+    $response = $soap->sendMessage($recipient, "messagemedia-php: please reply & press Enter on example when done.");
```
Comment (elruwen) [inline examples_replies.php]: Why adding press Enter? This is already mentioned below

Comment (shawr) [inline examples_replies.php]: makes more sense to me, but happy to remove it if you strongly disagree,

File: mmsoap/MMSoapClient.php [hunk 4]
```diff
@@ -38,6 +38,9 @@ public function __doRequest($request, $location, $action, $version, $one_way = 0
         $request = $dom->saveXML();
 
         $response = parent::__doRequest($request, $location, $action, $version);
+        // fix & bug which causes the following warning;
+        // PHP Warning:  DOMDocument::createElement(): unterminated entity reference 
+        $response = str_replace('&', '&amp;', $response);
```
Comment (soyang) [inline mmsoap/MMSoapClient.php]: Do we need to escape more strings? ', ", <, > ?

Comment (shawr) [inline mmsoap/MMSoapClient.php]: the others do not seem to break it
```
m4u-rob@m4urob-XPS-13-9343:/media/m4u-rob/b640c90f-43d7-4995-a61f-7858276323ca/oldpc/git/messagemedia-php-master$ php examples_replies.php 
** Do you want to send a message to +61417533586, y followed by ENTER to send, or ENTER to skip.

** Check for replies
Received 4 replies, 0 are remaining.
Reply #194040258 from 61491570156 received at 2016-04-11T15:05:40Z: reply 
Reply #194040259 from 61491570156 received at 2016-04-11T15:20:09Z: reply 2
Reply #194040260 from 61491570156 received at 2016-04-11T15:20:12Z: reply 3
Reply #194040261 from 61491570156 received at 2016-04-11T16:19:57Z: reply 4 &amp; stuff ? ! &lt; &gt; ' " 
** Do you want to confirm those replies, y followed by ENTER to send, or ENTER to skip.
n
```
what was in db 
```
mysql> select * from sms.replies order by received_Date desc limit 5;
+-----------+------------------+-------------+------------------------------------------------------------+------+---------------------+------------+--------+-------------+
| id        | account          | phone       | message                                                    | sent | received_date       | message_id | status | modem       |
+-----------+------------------+-------------+------------------------------------------------------------+------+---------------------+------------+--------+-------------+
| 194040261 | MessageUPtyLt884 | 61491570156 | reply 4 & stuff ? ! < > ' "                                | N    | 2016-04-12 02:19:57 |          0 | R      | unknown     |
| 194040260 | MessageUPtyLt884 | 61491570156 | reply 3                                                    | N    | 2016-04-12 01:20:12 |          0 | R      | unknown     |
| 194040259 | MessageUPtyLt884 | 61491570156 | reply 2                                                    | N    | 2016-04-12 01:20:09 |          0 | R      | unknown     |
| 194040258 | MessageUPtyLt884 | 61491570156 | reply                                                      | N    | 2016-04-12 01:05:40 |          0 | R      | unknown     |
| 194040257 | ArdorHealthSo041 | 13616483243 | Call 361-648-3245

That's another reference her names Lana | N    | 2015-08-21 02:00:19 |    1000995 | R      | 12012527981 |
+-----------+------------------+-------------+------------------------------------------------------------+------+---------------------+------------+--------+-------------+
5 rows in set (0.00 sec)


```

**Pull request merged**